lib/src/working_copy.rs

```diff
@@ -836,7 +836,9 @@ impl TreeState {
                                         .send((tracked_path.clone(), tree_value))
                                         .ok();
                                 }
-                                file_states_tx.send((tracked_path, new_file_state)).ok();
+                                if new_file_state != current_file_state {
+                                    file_states_tx.send((tracked_path, new_file_state)).ok();
```

Comment:
Since we are no longer indiscriminately sending updated file states, maybe we should rename this `updated_file_states_tx` to make it clear that we don't expect every file state to be reported?

Reply:
I don't think `file_states_tx` implies that unchanged states will also be sent. We do the same with `tree_entries_tx`. I prefer to leave it as is and it sounds like you don't feel strongly.